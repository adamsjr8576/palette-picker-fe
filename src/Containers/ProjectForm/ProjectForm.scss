.project-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  border: 3px solid black;
  padding: 15px;
  width: 80%;
  border-radius: 20px;
  height: 175px;
  background-color: #F2F2F2;
}

.project-name-label {
  font-size: 1.5em;
}

.input-make-project {
  width: 70%;
  font-size: 1em;
  padding: 10px;
  border-radius: 10px;
  outline: none;
  border: 2px solid black;
}

.project-form-error-container {
  width: 90%;
  height: 20px;
}

.form-error {
  margin: 0;
  color: red;
}

.save-project-btn {
  font-size: 1.25em;
  padding: 10px;
  border-radius: 10px;
  cursor: pointer;
  border: 0;
  background-color: #444343;
  color: white;
  outline: none;
}

.save-project-btn:hover {
  box-shadow: 3px 5px #203246;
}

.save-project-btn:focus {
  transform: scale(1.05);
}

.save-project-btn:active {
  transform: translate(3px, 3px);
  box-shadow: 1px 2px #203246;
}
