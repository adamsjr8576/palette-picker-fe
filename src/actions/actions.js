export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
})