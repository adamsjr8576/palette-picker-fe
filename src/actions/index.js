export const addNewPalette = currentPalette => ({
  type: 'ADD_PALETTE',
  currentPalette
});

export const updatePaletteLocked = hexCode => ({
  type: 'UPDATE_PALETTE_LOCKED',
  hexCode
});

export const addInitialProjects = projects => ({
  type: 'ADD_INITIAL_PROJECTS',
  projects
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTE_TO_PROJECT',
  palettes
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  project
});

export const deletePaletteFromStore = palette => ({
  type: 'DELETE_PALETTE',
  palette
});

export const deleteProjectFromStore = project => ({
  type: 'DELETE_PROJECT',
  project
});
