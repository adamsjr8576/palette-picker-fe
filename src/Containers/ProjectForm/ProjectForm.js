import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addProject } from '../../actions/actions';
import './ProjectForm.scss';

const ProjectForm = (props) => {

  const [ newProject, setNewProject] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    props.projects(newProject);
    resetInputs();
  }

  const resetInputs = () => {
    setNewProject('');
  }

  return(
    <form>
      <label for='make-project'>Make a new project:</label>
      <input 
        className='input-make-project'
        placeholder='Project NAme'
        maxLength='30'
        type='text'
        value={newProject}
        onChange={ e => setNewProject(e.target.value) }
      />
      <button type='button' className='save-project-btn' onClick={ e => handleSubmit(e) }>Save project</button>
    </form>
  )
}

export const mapDispatchToProps = dispatch => ({
  projects: (projectName) => dispatch(addProject(projectName))
})

export default connect(null, mapDispatchToProps)(ProjectForm);