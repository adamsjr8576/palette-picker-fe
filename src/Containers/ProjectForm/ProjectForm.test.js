import React from 'react';
import { shallow } from 'enzyme';
import { ProjectForm, mapDispatchToProps } from './ProjectForm';
import { addProject } from '../../actions/index';
import { render, fireEvent } from '@testing-library/react';

describe('ProjectForm', () => {

  let mockProps, setup

  beforeEach(() => {

    mockProps = {
      addProject: jest.fn()
    }

    setup = () => {
      const utils = render(<ProjectForm 
        addProject={mockProps.addProject}
      />)
      const input = utils.getByLabelText('name-project-input')
      const btn = utils.getByRole('button')
      return {
        input,
        btn,
        ...utils,
      }
    }
  });

  test('should match the snapshot', () => {
    const { utils } = setup();

    expect(utils).toMatchSnapshot();
  });

  test("should load with initial state of '' in the add project input", () => {
    const { input } = setup();
    const inputValue = input;

    expect(inputValue.textContent).toBe('');
  });
  
  test('should allow a user to input a name', () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: 'New Project Name'} });

    expect(input.value).toBe('New Project Name')
  });

  test('should reset the input when button is clicked', () => {
    const { input, btn } = setup();
    const inputValue = input;
    fireEvent.change(inputValue, { target: { value: 'New Project Name'} });
    fireEvent.click(btn);

    expect(inputValue.textContent).toBe('')
  });

});