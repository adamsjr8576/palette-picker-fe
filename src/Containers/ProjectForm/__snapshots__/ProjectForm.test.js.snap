// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm ProjectForm Component should match the snapshot 1`] = `undefined`;
